system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -0,0 +1,383 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+#define TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+
+#include <chrono>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rcpputils/asserts.hpp"
+
+#include "received_message_age.hpp"
+#include "received_message_period.hpp"
+#include "system_metrics_collector/constants.hpp"
+#include "system_metrics_collector/metrics_message_publisher.hpp"
+#include "topic_statistics_collector.hpp"
+
+using metrics_statistics_msgs::msg::MetricsMessage;
+
+namespace topic_statistics_collector
+{
+
+namespace
+{
+rcl_interfaces::msg::IntegerRange buildPositiveIntegerRange(
+  int64_t from,
+  int64_t to,
+  uint64_t step)
+{
+  rcl_interfaces::msg::IntegerRange positive_range;
+  positive_range.from_value = from;
+  positive_range.to_value = to;
+  positive_range.step = step;
+
+  return positive_range;
+}
```

Comment:
Were these moved from a .cpp file?
Now that these are in a .hpp file, the functions should be declared as `inline`. 

I'm unsure on the use of the anonymous namespace now that these functions are in a header file. I think being within an anonymous namespace will be redundant after adding `inline`. It might be worth keeping it around however to group the functions.

Reply:
Made them inline and removed the namespace (anonymous namespaces in headers are not recommended).